pkg/blobstore/cloud/s3_copy_mutator.go

```diff
@@ -0,0 +1,34 @@
+package cloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"github.com/buildbarn/bb-storage/pkg/clock"
+)
+
+type s3CopyMutator struct {
+	minRefreshAge time.Duration
+	clock         clock.Clock
+}
+
+// NewS3CopyMutator creates a BeforeCopyFunc which updates the "Used"
+// metadata field on s3 objects with the current time.
+func NewS3CopyMutator(minRefreshAge time.Duration, clock clock.Clock) BeforeCopyFunc {
+	return func(asFunc func(interface{}) bool) error {
+		var input *s3.CopyObjectInput
+		if !asFunc(&input) {
+			return fmt.Errorf("failed to get CopyObjectInput - not an s3 bucket?")
```

Comment:
I think this can only occur in case of actual programming mistakes, right? Maybe better to put a `panic()` here, so that we crash intentionally?

Reply:
Or API drift in gocloud, since we went through an interface. crash vs failing all operations is marginal, but I suppose there's no harm in a panic.